automation_script/ut_results_parse_automation.py

```diff
@@ -0,0 +1,402 @@
+#!/usr/bin/env python3
+
+"""This script parses PyTorch unit tests results based on configurations provided.
+The output will be written to automation_logs folder
+
+"""
+
+import argparse
+import os
+import shutil
+import subprocess
+from collections import namedtuple
+from datetime import datetime
+from pathlib import Path
+from parse_xml_results import (
+        parse_xml_report
+)
+from pprint import pprint
+
+# unit test status list
+UT_STATUS_LIST = [
+    "PASSED",
+    "MISSED",
+    "SKIPPED",
+    "FAILED",
+    "XFAILED",
+    "ERROR"
+]
+
+DEFAULT_CORE_TESTS = [
+    "test_nn",
+    "test_torch",
+    "test_cuda",
+    "test_ops",
+    "test_unary_ufuncs",
+    "test_autograd",
+    "inductor/test_torchinductor"
+]
+
+DISTRIBUTED_CORE_TESTS = [
+    "distributed/test_c10d_common",
+    "distributed/test_c10d_nccl",
+    "distributed/test_distributed_spawn"
+]
+
+# path variables
+test_shell_path = "/var/lib/jenkins/pytorch/.ci/pytorch/test.sh"
+test_run_test_path = "/var/lib/jenkins/pytorch/test/run_test.py"
+repo_test_log_folder_path = "/var/lib/jenkins/pytorch/automation_logs/"
+test_reports_src = "/var/lib/jenkins/pytorch/test/test-reports/"
```

Comment:
We should have PYTORCH_ROOT arg for the user which defaults to `/var/lib/jenkins/pytorch`